crates/core/src/factories/core_fetch/message_handler.rs

```diff
@@ -0,0 +1,212 @@
+use kitsune2_api::{
+    fetch::{
+        k2_fetch_message::FetchMessageType, K2FetchMessage, Request, Response,
+    },
+    transport::{TxBaseHandler, TxModuleHandler},
+    K2Error, K2Result, SpaceId, Url,
+};
+use prost::Message;
+use tokio::sync::mpsc::Sender;
+
+use super::{IncomingFetchRequest, IncomingResponse};
+
+#[derive(Debug)]
+pub(super) struct FetchMessageHandler {
+    pub(super) response_tx: Sender<IncomingFetchRequest>,
```

Comment:
This naming confuses me a little why are requests going into a response channel?

Reply:
It's also been quite confusing to me. I did a renaming sweep and am quite happy with it now. Let me know if you can understand it.

Reply:
Yeah, much easier to follow. Thank you!